pkg/configurations/configurations.go

```diff
@@ -0,0 +1,157 @@
+package configurations
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/fsnotify/fsnotify"
+
+	"github.com/thoas/go-funk"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	"github.com/spf13/viper"
+)
+
+const defaultInterfacesFilter = "veth*"
+const defaultNodeNetworkStateRefresh = 5
+
+var (
+	log        = logf.Log.WithName("configurations")
+	configPath string
+)
+
+// Takes os environment values
+func init() {
+	configPathTemp, isSet := os.LookupEnv("CONFIG_PATH")
+	if !isSet {
+		panic("CONFIG_PATH is mandatory")
+	}
+	configPath = configPathTemp
+}
+
+// Struct to match config file
+type Config struct {
+	NodeNetworkRefreshInterval int    `mapstructure:"node_network_state_refresh_interval"`
+	InterfaceFilter            string `mapstructure:"interfaces_filter"`
+}
+
+// Init defaults
+var c = &Config{
+	NodeNetworkRefreshInterval: defaultNodeNetworkStateRefresh,
+	InterfaceFilter:            defaultInterfacesFilter,
+}
+
+func GetCurrentConfig() Config {
+	return *c
+}
+
+var v = viper.New()
+
+// Updating relevant values with new config settings
+func (c *Config) updateConfig(v viper.Viper) {
+	log.Info("Updating configuration")
+	err := v.Unmarshal(&c)
+	log.Info(fmt.Sprintf("New configuration is %+v", c))
+	if err != nil {
+		log.Error(err, "Failed to unmarshal config file")
+		return
+	}
+}
+
+func exists(path string) bool {
+	_, err := os.Stat(path)
+	if err == nil {
+		return true
+	}
+	if os.IsNotExist(err) {
+		return false
+	}
+	return true
+}
+
+func ConfigWatcher() error {
+
+	ext := filepath.Ext(configPath)
+	if !funk.Contains(viper.SupportedExts, ext[1:]) {
+		return fmt.Errorf("file extension %s is not supported", ext)
+	}
+
+	configDir, _ := filepath.Split(configPath)
+	if !exists(configDir) {
+		return fmt.Errorf("folder %s doesn't exist, can't start configuration watcher", configDir)
+	}
+
+	v.SetConfigFile(configPath)
+	v.SetTypeByDefaultValue(true)
+
+	if err := v.ReadInConfig(); err != nil { // Find and read the config file
+		log.Info("Not able to read configuration, will update default values")
+	}
+
+	c.updateConfig(*v)
```

Comment:
Why do we need to updateConfig here? Won't it be done by the WatchConfig? This seems redundant

Reply:
It is the case where file already exists. Watch config will run updateConfig only on change